dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/EventValidator.java

```diff
@@ -0,0 +1,110 @@
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.tracker.validation.hooks;
+
+import static org.hisp.dhis.tracker.validation.hooks.All.all;
+import static org.hisp.dhis.tracker.validation.hooks.Each.each;
+import static org.hisp.dhis.tracker.validation.hooks.Field.field;
+import static org.hisp.dhis.tracker.validation.hooks.Seq.seq;
+
+import lombok.RequiredArgsConstructor;
+
+import org.hisp.dhis.tracker.bundle.TrackerBundle;
+import org.hisp.dhis.tracker.domain.Event;
+import org.hisp.dhis.tracker.validation.ValidationErrorReporter;
+import org.hisp.dhis.tracker.validation.Validator;
+import org.springframework.stereotype.Component;
+
+/**
+ * Validator to validate all {@link Event}s in the {@link TrackerBundle}.
+ */
+@RequiredArgsConstructor
+@Component( "org.hisp.dhis.tracker.validation.hooks.EventValidator" )
+public class EventValidator implements Validator<TrackerBundle>
+{
+    private final EventPreCheckUidValidator eventPreCheckUidValidator;
+
+    private final EventPreCheckExistenceValidator eventPreCheckExistenceValidator;
+
+    private final EventPreCheckMandatoryFieldsValidator eventPreCheckMandatoryFieldsValidator;
+
+    private final EventPreCheckMetaValidator eventPreCheckMetaValidator;
+
+    private final EventPreCheckUpdatableFieldsValidator eventPreCheckUpdatableFieldsValidator;
+
+    private final EventPreCheckDataRelationsValidator eventPreCheckDataRelationsValidator;
+
+    private final EventPreCheckSecurityOwnershipValidator eventPreCheckSecurityOwnershipValidator;
+
+    private final EventCategoryOptValidator eventCategoryOptValidator;
+
+    private final EventDateValidator eventDateValidator;
+
+    private final EventGeoValidator eventGeoValidator;
+
+    private final EventNoteValidator eventNoteValidator;
+
+    private final EventDataValuesValidator eventDataValuesValidator;
+
+    private final AssignedUserValidator assignedUserValidator;
+
+    private final RepeatedEventsValidator repeatedEventsValidator;
+
+    public Validator<TrackerBundle> eventValidator()
+    {
+        // @formatter:off
+        return all(
+                each(TrackerBundle::getEvents,
```

Comment:
Not critical, but codestyle (spaces after opening paranthesis) 😅 After 7 years they stick out so bad to me :P

Reply:
same below in "field("